flit/common.py

```diff
@@ -14,12 +15,20 @@ class Module(object):
     """
     def __init__(self, name, directory='.'):
         self.name = name
+        self.is_namespace_package = False
+        self.is_package = False
 
         # It must exist either as a .py file or a directory, but not both
+        name = (name + ".").split(".", 1)[0]
         pkg_dir = Path(directory, name)
         py_file = Path(directory, name+'.py')
         if pkg_dir.is_dir() and py_file.is_file():
             raise ValueError("Both {} and {} exist".format(pkg_dir, py_file))
+        elif pkg_dir.is_dir() and "." in self.name:
+            self.path = pkg_dir
+            self.ns_path = Path(directory, self.name.replace('.', os.sep))
```

Comment:
I might request that some of these names change a bit - to me, the 'namespace path' would refer to the outer directory, not the inner one. But I haven't worked out what names I want yet.

The code should also check that the inner directory exists - currently I think your changes make it possible to instantiate this object when the directory is missing, which wasn't possible before.